WordPress/Classes/Models/Notifications/NotificationRange.swift

```diff
@@ -41,19 +41,36 @@ class NotificationRange {
     /// Designated Initializer
     ///
     init?(dictionary: [String: AnyObject]) {
-        guard let type = dictionary[RangeKeys.RawType] as? String, let indices = dictionary[RangeKeys.Indices] as? [Int],
-            let start = indices.first, let end = indices.last else {
+        guard let indices = dictionary[RangeKeys.Indices] as? [Int],
+            let start = indices.first,
+            let end = indices.last
+        else {
             return nil
         }
 
-        kind = Kind(rawValue: type) ?? .Site
         range = NSMakeRange(start, end - start)
         siteID = dictionary[RangeKeys.SiteId] as? NSNumber
 
         if let rawURL = dictionary[RangeKeys.URL] as? String {
             url = URL(string: rawURL)
         }
 
+        // The Duck is STILL ALIVE:
+        // ========================
+        // I truly hope the reviewer, and the Opensource Community can forgive this ongoing hack.
+        // Notifications can now carry URL's without specifying an explicit 'Type'. For that reason,
+        // surprise!... we're now also inferring the Notification Type.
+        //
+        if let type = dictionary[RangeKeys.RawType] as? String, let theKind = Kind(rawValue: type) {
+            kind = theKind
+        } else if siteID != nil {
+            kind = .Site
+        } else if url != nil {
+            kind = .Link
+        } else {
+            return nil
+        }
```

Comment:
no swift expert here, but by the time it's getting this complex, why not create a separate function to return the note type. seems like it could save some headache in making the inference rules easier to follow

```swift
getType( dictionary: [String: AnyObject] ) {
	// complex logic clearly here with early aborts

	// Site ranges have a siteId
	if ( siteId != nil ) {
		return .Site
	}
}
```

Reply:
Honestly, i truly wanted to do precisely that. But the deal is: 

the Type is tied up to other values parsed a couple lines above, which translates into having a method that returns the Type, based on two (optional) parameters, and a dictionary.

(Which IMHO may look even uglier than having that if in the initializer).